src/__tests__/GuestNameInput.test.js

```diff
@@ -0,0 +1,59 @@
+import "@testing-library/jest-dom";
+import "@testing-library/user-event";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event/";
+import GuestNameInput from "../components/GuestNameInput";
+
+describe("input field requires valid usernames", () => {
+  it("should accept regular usernames", async () => {
+    render(
+      <form data-testid="form">
+        <GuestNameInput />
+      </form>
+    );
+    const form = screen.getByTestId("form");
+    const inputfield = screen.getByRole("textbox");
+    await userEvent.type(inputfield, "Sarah");
+    expect(form).toBeValid();
+  });
+
+  it("should not accept empty usernames", async () => {
+    render(
+      <form data-testid="form">
+        <GuestNameInput />
+      </form>
+    );
+    const form = screen.getByTestId("form");
+    const inputfield = screen.getByRole("textbox");
+    await userEvent.type(inputfield, "  ");
+    expect(form).not.toBeValid();
+  });
+
+  it("should require an input", async () => {
+    render(
+      <form data-testid="form">
+        <GuestNameInput />
+      </form>
+    );
+    const inputfield = screen.getByRole("textbox");
+    expect(inputfield).toBeRequired();
+  });
+
+  it("should not allow input of usernames with more than 40 letters", async () => {
+    render(
+      <form data-testid="form">
+        <GuestNameInput />
+      </form>
+    );
+    const inputfield = screen.getByRole("textbox");
+
+    await userEvent.type(
+      inputfield,
+      "asdfubsafubasdfiusabfiusdffsufbisaudfbsf asiudfbasdufbaisdufbaisudfbaisubfiusabdf"
+    );
+    //the expected value below is exactly 40 letters:
+    expect(inputfield).toHaveDisplayValue(
+      "asdfubsafubasdfiusabfiusdffsufbisaudfbsf"
+    );
```

Comment:
Alternatively, you could try to figure out if there's a more direct way of getting access to the length of the input field's value. Since you're not interested in the exact representation of the string that was entered, you needn't check it for its accuracy but its length.

Reply:
Yes, I found the more direct way 😅 thanks 👍